src/browser/pages/vscode.ts

```diff
@@ -160,9 +156,72 @@ export function setBodyBackgroundToThemeBackgroundColor(document: Document, loca
   return null
 }
 
-try {
+export function registerLoadBundleOnNlsConfig(nlsConfig: NlsConfiguration, base: string) {
+  const bundles: {
+    [key: string]: string
+  } = {}
+
+  type LoadBundleCallback = (_: undefined, result?: string) => void
+
+  nlsConfig.loadBundle = (bundle: string, _language: string, cb: LoadBundleCallback): void => {
+    const result = bundles[bundle]
+    if (result) {
+      return cb(undefined, result)
+    }
+    // FIXME: Only works if path separators are /.
+    const path = nlsConfig._resolvedLanguagePackCoreLocation + "/" + bundle.replace(/\//g, "!") + ".nls.json"
+    fetch(`${base}/vscode/resource/?path=${encodeURIComponent(path)}`)
+      .then((response) => response.json())
+      .then((json) => {
+        bundles[bundle] = json
+        cb(undefined, json)
+      })
+      .catch(cb)
+  }
+}
+
+/**
+ * A helper function to encapsulate all the
+ * logic used in this file.
+ *
+ * It does the following:
+ * - grabs the options using getOptions
+ * - gets the nlsConfig
+ * - registers loadBundle on nlsConfig
+ * - gets the loader
+ * - sets loader on self.require
+ * - sets the body background color to match the theme
```

Comment:
Minor nit, we might want to avoid comments that duplicate what the code is doing; they tend to become out of date quickly (someone updates the code but not the comment, etc). :sweat_smile:  

On another note, I think `main` is a pretty common pattern (at least in our repo it is) so feel free to avoid the justifications if you want! :smile:

Reply:
Good point! I guess I tend to overcomment things but I can see why this might not be a good idea here. I'll slim down!